Alpha6 updates and Mapping Fixes

rickcasson opened the pull request:
This updates to the Plugin to the latest version of the Health Connect API, fixes various mapping issues for the different Record types, adds granularity  to the `hasPermissions` and `requestPermissions` methods, and adds the `revokePermissions` method .

Things to note: 
- ~I've been having issue running on device with Android 14 (SDK 34) installed, but it seems to be a common problem with other Health Connect plugins too.~ (edit: this is now fixed [based on a PR](https://github.com/cph-cachet/flutter-plugins/pull/834) that @eliasteeny made to the health plugin)
- the previous usage of `hasPermissions` and `requestPermissions` will still work, but you can now pass lists of record types that need read, write, or both types of permissions (see updated example for details)
- This fixes the issue flagged in #16 
- if you are using ProGuard, you will need to add to your proguard-rules file:
```
-keep public class androidx.health.** { *; }
-dontwarn androidx.health.**
```

rickcasson committed:
update hc version and fix issues

rickcasson committed:
update alpha version

rickcasson committed:
update gradle and pull out stuff that was breaking

rickcasson committed:
fix offset parsing

rickcasson committed:
need to parse duration from zone offset

rickcasson committed:
Revert "need to parse duration from zone offset"

This reverts commit 389c1a72143c5f63a73b211ce56e1400b17496e6.

rickcasson committed:
add internal duration parser

rickcasson committed:
Update blood_glucose_record.dart

rickcasson committed:
fix mapping

rickcasson committed:
fix assertion

rickcasson committed:
Update blood_glucose_record.dart

rickcasson committed:
fix mapping

rickcasson committed:
testing casting filter

rickcasson committed:
fix datetime parsing

rickcasson committed:
remove logging

rickcasson committed:
remove package info

rickcasson committed:
remove print

rickcasson committed:
update ChangeLog

rickcasson committed:
fix blood pressure mapping

rickcasson committed:
fix unit conversions

rickcasson committed:
Update FlutterHealthConnectPlugin.kt

rickcasson committed:
fixing mapping

rickcasson committed:
fix type mapping

rickcasson committed:
fix other maps

rickcasson committed:
add delete records logs

rickcasson committed:
just reply true

rickcasson committed:
clean up logs

rickcasson committed:
Fixes to support Android 14

chadpav commented:
@rickcasson I added updated to the README in @eliasteeny PR. I think you will want to include those here. 
https://github.com/cph-cachet/flutter-plugins/pull/834/commits/cd71d0fe21dc7361548dd8325c1a500898ec2bfd

Did this PR also fix the issues he found with requesting SLEEP data?

rickcasson committed:
refactor timezone offset parsing

rickcasson committed:
Remove old sleep stage record

rickcasson committed:
Add more control over permission types

rickcasson committed:
Update example and readme

rickcasson committed:
roll back gradle changes and only include used permissions in AndroidManifest

rickcasson committed:
Added revokeAllPermissions method and update readme

thildebrand1 committed:
Fixed key / value issue in sleep stage map - key is 'stage', not 'type'.

thildebrand1 committed:
Update sleep_session_record.dart

For clarity, we still want this named type - just the map key changed to 'stage'.

rickcasson committed:
Merge pull request #1 from thildebrand1/alpha6-updates

Updated sleep stage map key to correctly retrieve and set sleep stages.

rickcasson committed:
rename stage variable for consistancy

rickcasson committed:
fix has permission crash

rickcasson committed:
fix startTime assert to allow for same endTime

rickcasson committed:
fix carbohydrates mapping
